### ISSUE TITLE
Pass the parameters in the url as properties when the connection is created

### ISSUE THREAD

**Issue by DavideD** [hibernate/hibernate-reactive#864]
Each Vert.x Driver accept different additional parameter keys in the URL.
For example, in [PostgreSQL](https://vertx.io/docs/vertx-pg-client/java/#_connection_uri):
```
host
hostaddr
port
user
password
dbname
sslmode
properties including(application_name, fallback_application_name, search_path)
```

I think that when we create the `SqlClientOptions` in [DefaultSqlClientPoolConfiguration](https://github.com/hibernate/hibernate-reactive/blob/26e0aef99edaef45f29b3ee985dbe976b4e72b73/hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/DefaultSqlClientPoolConfiguration.java#L151) we don't keep track of them.

We should test this.

**Comment by blafond** [comment]
@DavideD  We currently don't capture any properties besides user or pwd.  Is there any reason we shouldn't capture other properties and add these custom properties to SqlClientOptions?  Seems fairly straight forward.

**Comment by DavideD** [comment]
No reason @blafond. I think that's what we should do

**Comment by DavideD** [comment]
@blafond I've reassined this issue to you. It's related to what are you doint right now, isn't it?

**Comment by blafond** [comment]
yes

**Comment by blafond** [comment]
I addressed this in #1059 and added tests for vertx db clients to verify that default props (example: see [PgConnectOptions](https://github.com/eclipse-vertx/vertx-sql-client/blob/0d10efe2bbedea3c27b380d69add2f1e036da9ad/vertx-pg-client/src/main/java/io/vertx/pgclient/PgConnectOptions.java#L122))

Note that any other URL properties not defined in `SqlConnectOptions` implementations are currently ignored in vertx-sql-client (See: [#1115](https://github.com/eclipse-vertx/vertx-sql-client/issues/1115)). I have a [branch](https://github.com/blafond/vertx-sql-client/commit/b6634e25252d50d7f2a2fc96337bb7adebc4dbe1) I've tested which logs a WARNING when unsupported properties are discovered in the `SqlConnectOptions.fromUri()` call

**Comment by pendula95** [comment]
@DavideD 
I run on unexpected behavior while trying to make @ColumnTransformer that uses function pgp_sym_encrypt from added extension `pgcrypto`
I expected that correct search_path was propagated from hibernate.default_schema but looks like this is not the case.
Schema for this function is not propagated and it fails to find it as search_path is always wrong and not respecting the configuration param.
After checking with vertx pg client I see that it has a correct param search_path that can be set via params or via url.
So I try to do this by specifying ?search_path=xyz on the end of my jdbc url. This does not work also and the  query I used to debug before `show search_path` again gives wrong value.
Then I checked how hibernate handles custom params and I see this really strange line that will check additional params only if username or password are null. 
https://github.com/hibernate/hibernate-reactive/blob/f17e8a1e80c70329baaf3d98fc7f229bf68b02e7/hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/DefaultSqlClientPoolConfiguration.java#L111-L117

Because I use hybrid approach where only host information is passed via jdbcurl and username/password/schema are passed via property values this additional values like search_path are totally ignored.

So from here I think there are 2 issues:

1) `hibernate.default_schema` is not propagated and used with fucntions that need to be looked up in correct schema.
2) Additional connection params are ignored when username and password are passed as properties.

After tweaking my tests that so that username is passed via param, password via url, rest of the params like (search_path) are respected and my @ColumnTransformer works as expected and this function is found.

In order to reproduce the issue:
create pg database
delete public schema
create some other schema
CREATE EXTENSION pgcrypto
try to use any of  pgcrypto functions

I have simple reproducers for all but just by looking at the code for parsing of params the issue 2) is simple to fix and reproduce.
Issue 1 I have not looked at code base so I am not sure If I am even right if schema should be propagated to search_path but I can open up another thread for this.
